addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanner.java

```diff
@@ -0,0 +1,147 @@
+                                            Alert.CONFIDENCE_HIGH,
+                                            getName());
+
+                            alert.setDetail(
+                                    getString("desc"),
+                                    msg.getRequestHeader().getURI().toString(),
+                                    "",
+                                    "",
+                                    "",
+                                    getString("soln"),
+                                    getString("refs"),
+                                    element.toString(),
+                                    16, // CWE CATEGORY: Configuration
+                                    15, // Application Misconfiguration
+                                    msg);
+                            parent.raiseAlert(id, alert);
+                        });
+    }
+
+    private static Predicate<Element> unsafeSubResource(String origin) {
+        return element -> {
+            Optional<String> maybeHostname = SupportedElements.getHost(element);
+            return element.getAttributeValue("integrity") == null
+                    && !maybeHostname.map(hostname -> hostname.matches(origin)).orElse(false);
+        };
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public String getName() {
+        return getString("name");
+    }
+
+    private static String getString(String param) {
+        return Constant.messages.getString(MESSAGE_PREFIX + param);
+    }
+
+    @Override
+    public int getPluginId() {
+        return 90003;
+    }
+
+    private enum SupportedElements {
+        // From
+        // https://w3c.github.io/webappsec-subresource-integrity/#verification-of-html-document-subresources
+        // To support integrity metadata for some of these elements, a new integrity attribute is
+        // added
+        // to the list of content attributes for the link and script elements.
+        // Note: A future revision of this specification is likely to include integrity support for
+        // all
+        // possible subresources, i.e., a, audio, embed, iframe, img, link, object, script, source,
+        // track, and video elements.
+
+        SCRIPT(HTMLElementName.SCRIPT, "src"),
+        LINK(HTMLElementName.LINK, "href");
+
+        final String tag;
+        final String attribute;
+
+        SupportedElements(String tag, String attribute) {
+            this.tag = tag;
+            this.attribute = attribute;
+        }
+
+        public static boolean contains(String tag) {
+            return Stream.of(values()).anyMatch(e -> tag.equals(e.tag));
+        }
+
+        public static Optional<String> getHost(Element element) {
+            String url =
+                    element.getAttributeValue(
+                            SupportedElements.valueOf(element.getName().toUpperCase()).attribute);
```

Comment:
Should use `Locale.ROOT` to not rely on default locale (e.g. `i` would cause problems in some locales).